### PR TITLE
Add category model

### PR DESCRIPTION
**Pull request by OlegAvanesov**

### CODE REVIEW

File: src/main/java/mate/academy/mainspringproject/mappers/CategoryMapper.java [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+
+import mate.academy.mainspringproject.config.MapperConfig;
+import mate.academy.mainspringproject.dto.category.CategoryRequestDto;
+import mate.academy.mainspringproject.dto.category.CategoryResponseDto;
+import mate.academy.mainspringproject.model.Category;
+import org.mapstruct.Mapper;
+
+@Mapper(config = MapperConfig.class)
+public interface CategoryMapper {
+
```
Comment (okuzan) [inline src/main/java/mate/academy/mainspringproject/mappers/CategoryMapper.java:11]:
```suggestion
```

File: src/main/java/mate/academy/mainspringproject/mappers/CategoryMapper.java [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+    CategoryResponseDto toDto(Category category);
+
+    Category toEntity(CategoryRequestDto categoryDto);
+
```
Comment (okuzan) [inline src/main/java/mate/academy/mainspringproject/mappers/CategoryMapper.java:15]:
```suggestion
```

File: src/main/java/mate/academy/mainspringproject/model/Book.java [hunk 3]
```diff
@@ -33,4 +39,13 @@ public class Book {
     private String coverImage;
     @Column(name = "is_deleted")
     private Boolean isDeleted = false;
```
Comment (okuzan) [inline src/main/java/mate/academy/mainspringproject/model/Book.java:41]:
```suggestion
    private boolean isDeleted = false;
```
I'm not sure NULL would make sense here

File: src/main/java/mate/academy/mainspringproject/model/Category.java [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+import jakarta.persistence.Table;
+import lombok.Data;
+import org.hibernate.annotations.SQLDelete;
+import org.hibernate.annotations.SQLRestriction;
+
+@Entity
+@Data
+@Table(name = "categories")
+@SQLDelete(sql = "UPDATE categories SET is_deleted = true WHERE id=?")
+@SQLRestriction("is_deleted = false")
```
Comment (okuzan) [inline src/main/java/mate/academy/mainspringproject/model/Category.java:17]: TRUE and FALSE

File: src/main/java/mate/academy/mainspringproject/model/Category.java [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+import jakarta.persistence.Id;
+import jakarta.persistence.Table;
+import lombok.Data;
+import org.hibernate.annotations.SQLDelete;
+import org.hibernate.annotations.SQLRestriction;
+
+@Entity
+@Data
+@Table(name = "categories")
+@SQLDelete(sql = "UPDATE categories SET is_deleted = true WHERE id=?")
```
Comment (okuzan) [inline src/main/java/mate/academy/mainspringproject/model/Category.java:16]:
```suggestion
@SQLDelete(sql = "UPDATE categories SET is_deleted = true WHERE id = ?")
```

File: src/main/java/mate/academy/mainspringproject/repository/book/BookRepository.java [hunk 6]
```diff
@@ -1,8 +1,16 @@
...
+import java.util.List;
 import mate.academy.mainspringproject.model.Book;
+import org.springframework.data.domain.Pageable;
 import org.springframework.data.jpa.repository.JpaRepository;
 import org.springframework.data.jpa.repository.JpaSpecificationExecutor;
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.data.repository.query.Param;
 
 public interface BookRepository extends JpaRepository<Book, Long>, JpaSpecificationExecutor<Book> {
+
```
Comment (okuzan) [inline src/main/java/mate/academy/mainspringproject/repository/book/BookRepository.java:12]:
```suggestion
```

File: src/main/java/mate/academy/mainspringproject/service/user/UserServiceImpl.java [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+        Role role = getRoleByName(Role.RoleName.ROLE_USER);
+        userForRegistration.setRoles(Set.of(role));
+
+        return userMapper.toDto(userRepository.save(userForRegistration));
+    }
+
+    private Role getRoleByName(Role.RoleName roleName) {
+        return roleRepository.findAll().stream()
+                .filter(role -> role.getName().equals(roleName))
+                .findFirst().orElseThrow(
```
Comment (okuzan) [inline src/main/java/mate/academy/mainspringproject/service/user/UserServiceImpl.java:40]:
```suggestion
                .findFirst()
                .orElseThrow(
```

File: src/main/java/mate/academy/mainspringproject/service/category/CategoryService.java [hunk 8]
```diff
@@ -0,0 +1,19 @@
+package mate.academy.mainspringproject.service.category;
+
+import java.util.List;
+import mate.academy.mainspringproject.dto.category.CategoryRequestDto;
+import mate.academy.mainspringproject.dto.category.CategoryResponseDto;
+import org.springframework.data.domain.Pageable;
+
+public interface CategoryService {
+
```
Comment (okuzan) [inline src/main/java/mate/academy/mainspringproject/service/category/CategoryService.java:9]:
```suggestion
```

**Review by okuzan — CHANGES_REQUESTED** [review]
minor fixes

**Review by MentorMA — APPROVED** [review]

**Pull request merged**